src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Net46.cs

```diff
@@ -0,0 +1,1193 @@
+                if (string.Compare(currentHeader, HttpKnownHeaderNames.ContentLength,
+                    StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    continue;
+                }
+
+                string[] values = webResponseHeaders.GetValues(i);
+
+                if (!responseHeaders.TryAddWithoutValidation(currentHeader, values))
+                {
+                    bool result = contentHeaders.TryAddWithoutValidation(currentHeader, values);
+                    // WebHeaderCollection should never return us invalid header names.
+                    Contract.Assert(result, "Invalid header name.");
+                }
+            }
+
+            return response;
+        }
+
+        private void HandleAsyncException(RequestState state, Exception e)
+        {
+            // Use 'SendAsync' as method name, since this method is only called by methods in the async code path. Using
+            // 'SendAsync' as method name helps relate the exception to the operation in log files.
+            if (Logging.On) Logging.Exception(Logging.Http, this, "SendAsync", e);
+
+            // If the WebException was due to the cancellation token being canceled, throw cancellation exception.
+            if (state.cancellationToken.IsCancellationRequested)
+            {
+                state.tcs.TrySetCanceled();
+            }
+            // Wrap expected exceptions as HttpRequestExceptions since this is considered an error during 
+            // execution. All other exception types, including ArgumentExceptions and ProtocolViolationExceptions
+            // are 'unexpected' or caused by user error and should not be wrapped.
+            else if (e is WebException || e is IOException)
+            {
+                state.tcs.TrySetException(new HttpRequestException(SR.net_http_client_execution_error, e));
+            }
+            else
+            {
+                state.tcs.TrySetException(e);
+            }
+        }
+
+        private static void OnCancel(object state)
+        {
+            HttpWebRequest webRequest = state as HttpWebRequest;
+            Contract.Assert(webRequest != null);
+
+            webRequest.Abort();
+        }
+
+        #endregion Request Processing
+
+        #region Helpers
+
+        private void SetOperationStarted()
+        {
+            if (!_operationStarted)
+            {
+                _operationStarted = true;
+            }
+        }
+
+        private void CheckDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(GetType().FullName);
+            }
+        }
+
+        internal void CheckDisposedOrStarted()
+        {
+            CheckDisposed();
+            if (_operationStarted)
+            {
+                throw new InvalidOperationException(SR.net_http_operation_started);
+            }
+        }
+
+        private static bool AreEqual(string x, string y)
```

Comment:
There were three or four places earlier where string.Compare was used directly; should they be changed to use this AreEqual?  The body here could also be changed to use string.Equal(...) instead of string.Comare(...) == 0.
https://msdn.microsoft.com/en-us/library/t4411bks(v=vs.110).aspx